src/TritonRoute/src/dr/FlexGridGraph_maze.cpp

```diff
@@ -802,9 +802,21 @@ frMIdx FlexGridGraph::getUpperBoundIndex(const frVector<frCoord>& tracks,
       ||  // comment out for non-buffer enablement
       currGrid.getLastDir() == dir) {
     return false;
-  } else {
-    return true;
+  } 
+  if (ndr_) {
+      frCoord hw = (frCoord)design_->getTech()->getLayer(getLayerNum(currGrid.z()))->getWidth();
+      if (ndr_->getWidth(currGrid.z()) > hw && !isSrc(currGrid.x(), currGrid.y(), currGrid.z())) {
+          hw = ndr_->getWidth(currGrid.z())/2;
+          if ((dir == frDirEnum::N || dir == frDirEnum::S) && (xCoords_[currGrid.x()]-hw < design_->getTopBlock()->getDieBox().left() || 
+              xCoords_[currGrid.x()]+hw > design_->getTopBlock()->getDieBox().right()) ||
+              (dir == frDirEnum::E || dir == frDirEnum::W) && (yCoords_[currGrid.y()]-hw < design_->getTopBlock()->getDieBox().bottom() ||
+              yCoords_[currGrid.y()]+hw > design_->getTopBlock()->getDieBox().top())) {
+              return false;
```

Comment:
Since this conditional is so long adding a comment above explaining roughly what it's trying to check for would help with readability.